dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java

```diff
@@ -179,6 +177,21 @@ public boolean doesUserHaveAccessToPortlet(final String portletId, final User us
 		if(loadLayoutsForUser(user).stream(). anyMatch(layout -> layout.getPortletIds().contains(portletId))){
 			return true;
 		}
+		if(portletId.equals("edit-page") ){
+			final RoleAPI roleAPI = APILocator.getRoleAPI();
+			final List<Role> foundRoles = roleAPI.loadRolesForUser( user.getUserId(), false );
+			final PermissionAPI permAPI = APILocator.getPermissionAPI();
+
+			for (final Role role: foundRoles){
+				final List<Permission> perms = permAPI.getPermissionsByRole(role, true, true);
+				for(final Permission perm : perms){
+					if((perm.getType().equals(IHTMLPage.class.getCanonicalName()) || perm.getType().equals(Contentlet.class.getCanonicalName()))
+							&& (perm.getPermission() == PermissionAPI.PERMISSION_EDIT || perm.getPermission() == 3)){
+						return true;
+					}
+				}
+			}
```

Comment:
maybe is better move this code into a local method, something like

```suggestion
if(portletId.equals("edit-page") ){
    localMethod(...);
}
```

I think is a readable code

Reply:
Thanks for the feedback, change included https://github.com/dotCMS/core/pull/26447/commits/a389d706ea855f1c12c2c8181429517022663a4b